server/protocols/jmap/src/main/java/org/apache/james/jmap/JMAPServer.java

```diff
@@ -52,17 +64,102 @@ public Port getPort() {
     }
 
     public void start() {
+        ImmutableListMultimap<Endpoint, JMAPRoute> collect = jmapRoutes.stream()
+            .flatMap(JMAPRoutes::routes)
+            .collect(Guavate.toImmutableListMultimap(JMAPRoute::getEndpoint));
+
         if (configuration.isEnabled()) {
             server = Optional.of(HttpServer.create()
                 .port(configuration.getPort()
                     .map(Port::getValue)
                     .orElse(RANDOM_PORT))
-                .route(routes -> jmapRoutes.forEach(jmapRoute -> jmapRoute.define(routes)))
+                .route(routes -> jmapRoutes.forEach(jmapRoute -> collect.asMap().forEach(
+                    (endpoint, route) -> injectRoutes(routes, endpoint, route))))
                 .wiretap(configuration.wiretapEnabled())
                 .bindNow());
         }
     }
 
+    private HttpServerRoutes injectRoutes(HttpServerRoutes builder, Endpoint endpoint, Collection<JMAPRoute> routesList) {
+        if (routesList.size() == 1) {
+            JMAPRoute next = routesList.iterator().next();
+            switch (endpoint.getVerb()) {
+                case GET:
+                    builder.get(endpoint.getPath(), (req, res) ->
+                        getExistingRoute(extractRequestVersionHeader(req), next).apply(req, res));
+                    break;
+                case POST:
+                    builder.post(endpoint.getPath(), (req, res) ->
+                        getExistingRoute(extractRequestVersionHeader(req), next).apply(req, res));
+                    break;
+                case DELETE:
+                    builder.delete(endpoint.getPath(), (req, res) ->
+                        getExistingRoute(extractRequestVersionHeader(req), next).apply(req, res));
+                    break;
+                case OPTIONS:
+                    builder.options(endpoint.getPath(), (req, res) ->
+                        getExistingRoute(extractRequestVersionHeader(req), next).apply(req, res));
+                    break;
+            }
+        } else if (routesList.size() == 2) {
+            ImmutableList<JMAPRoute> sorted = routesList.stream()
+                .sorted(Comparator.comparing(JMAPRoute::getVersion))
+                .collect(Guavate.toImmutableList());
+            JMAPRoute draftRoute = sorted.get(0);
+            JMAPRoute rfc8621Route = sorted.get(1);
+
+            switch (endpoint.getVerb()) {
```

Comment:
Idem that switch case is evil, previous suggestions will allow getting rid of it